src/xopen/__init__.py

```diff
@@ -614,9 +652,42 @@ def _detect_format_from_extension(filename: Union[str, bytes]) -> Optional[str]:
     return None
 
 
+def _file_or_path_to_binary_stream(
+    file_or_path: FileOrPath, binary_mode: str
+) -> Tuple[BinaryIO, bool]:
+    if binary_mode not in ("rb", "wb", "ab"):
+        raise AssertionError()
+    if file_or_path == "-":
+        return _open_stdin_or_out(binary_mode), False
+    if isinstance(file_or_path, (str, bytes)) or hasattr(file_or_path, "__fspath__"):
+        return open(os.fspath(file_or_path), binary_mode), True  # type: ignore
+    if isinstance(file_or_path, (io.BufferedReader, io.BufferedWriter)):
+        return file_or_path, False
+    if isinstance(file_or_path, io.TextIOWrapper):
+        return file_or_path.buffer, False
+    if isinstance(file_or_path, io.IOBase) and not hasattr(file_or_path, "encoding"):
+        # Text files have encoding attributes. This file is binary:
+        return file_or_path, False
+    else:
```

Comment:
For consistency, either do not use an `else` here an dedent the `raise` or use `elif` for the preceding `if`s.